portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -88,17 +93,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = getParameterOrDefault(request, "comment", /*default comment value=*/"");
-    long timestamp = System.currentTimeMillis();
-    
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("comment", comment);
-    commentEntity.setProperty("timestamp", timestamp);
-    
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(commentEntity);
-
-    response.sendRedirect("/index.html");
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String comment = getParameterOrDefault(request, "comment", /*default comment value=*/"");
+      long timestamp = System.currentTimeMillis();
+      
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("comment", comment);
+      commentEntity.setProperty("timestamp", timestamp);
+      
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+
+      response.sendRedirect("/index.html");
```

Comment:
Some kind of success message would be nice. Failure too.

Reply:
I've added a failure response. Not sure if I did this right but I think it is clear enough. If this is OK I will add some more failure message handling in future branches.